de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/navigation/proxy/SelectFolderByPathLoadHandler.java

```diff
@@ -66,24 +68,36 @@
                                            .trimResults()
                                            .omitEmptyStrings()
                                            .split(folderToSelect.getPath()));
-
-        rootsLoaded = navigationPresenter.rootsLoaded();
-        if (rootsLoaded) {
-            initPathsToLoad();
-        }
     }
 
-    public SelectFolderByPathLoadHandler(final HasPath folderToSelect,
-                                         final NavigationView.Presenter navigationPresenter,
-                                         final NavigationView.Presenter.Appearance appearance,
-                                         final IsMaskable maskable,
-                                         final IplantAnnouncer announcer) {
-
-        this(folderToSelect, navigationPresenter, appearance, maskable, announcer, null);
+    SelectFolderByPathLoadHandler(final HasPath folderToSelect,
+                                  final NavigationView.Presenter navigationPresenter,
+                                  final RefreshFolderSelected.RefreshFolderSelectedHandler refreshHandler,
+                                  final NavigationView.Presenter.Appearance appearance,
+                                  final IsMaskable maskable,
+                                  final IplantAnnouncer announcer) {
+        this(folderToSelect, navigationPresenter, refreshHandler, appearance, maskable, announcer, null);
     }
 
-    public void setHandlerRegistration(HandlerRegistration handlerRegistration) {
-        this.handlerRegistration = handlerRegistration;
+    public static void registerFolderLoader(final HasPath folderToSelect,
+                                            final NavigationView.Presenter navigationPresenter,
+                                            final RefreshFolderSelected.RefreshFolderSelectedHandler refreshHandler,
+                                            final NavigationView.Presenter.Appearance appearance,
+                                            final IsMaskable maskable,
+                                            final IplantAnnouncer announcer,
+                                            HasLoadHandlers<Folder, List<Folder>> loader) {
```

Comment:
This method has 0% test coverage. Since it is important that the registration occur before initiation of lazy-loading logic, it would be prudent to put a test to it.